packages/grafana-ui/src/types/icon.ts

```diff
@@ -226,17 +226,6 @@ export function getFieldTypeIcon(field?: Field): IconName {
   return 'question-circle';
 }
 
-function isValidIconName(iconName: string): iconName is IconName {
-  const namedIcons: string[] = availableIcons;
-  const brandIcons: string[] = avaibleBrandIcons;
-
-  return namedIcons.includes(iconName) || brandIcons.includes(iconName);
-}
-
-export function toIconName(iconName: string): IconName | undefined {
-  if (isValidIconName(iconName)) {
-    return iconName;
-  }
-
-  return undefined;
+export function toIconName(iconName: string): IconName {
+  return iconName as IconName;
```

Comment:
_if_ we allow arbitrary strings as icons, then this function just becomes a no-op. It can be removed (or in the interim just return the string input without any type assertions)